### PR TITLE
#26 Change from AlarmManager to JobInfo/JobScheduler

### PR DESCRIPTION
**Pull request by jogrimst**

### CODE REVIEW

File: app/src/main/java/org/literacyapp/appstore/receiver/BootReceiver.java [hunk 1]
```diff
@@ -1,38 +1,33 @@
...
-        calendarDaily.set(Calendar.HOUR_OF_DAY, 15);
-        if (Calendar.getInstance().get(Calendar.HOUR_OF_DAY) >= 15) {
-            // Make sure the alarm doesn't trigger until tomorrow
-            calendarDaily.add(Calendar.DAY_OF_MONTH, 1);
+        Log.i(getClass().getName(), "onReceive");
+
+        // Initiate background job for synchronizing applications
+        ComponentName componentName = new ComponentName(context, ApplicationSynchronizationService.class);
+        JobInfo.Builder builder = new JobInfo.Builder(1, componentName);
+        builder.setRequiresDeviceIdle(true);
```
Comment (jogrimst) [inline app/src/main/java/org/literacyapp/appstore/receiver/BootReceiver.java]: This requires the display to be switched off.

> Idle mode is a loose definition provided by the system, which means that the device is not in use, and has not been in use for some time. As such, it is a good time to perform resource heavy jobs. Bear in mind that battery usage will still be attributed to your application, and surfaced to the user in battery stats.

File: app/src/main/java/org/literacyapp/appstore/service/ApplicationSynchronizationService.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.literacyapp.appstore.util.ConnectivityHelper;
+
+public class ApplicationSynchronizationService extends JobService {
+
+    @Override
+    public boolean onStartJob(JobParameters jobParameters) {
+        Log.i(getClass().getName(), "onStartJob");
+
+        boolean isWifiEnabled = ConnectivityHelper.isWifiEnabled(getApplicationContext());
+        Log.i(getClass().getName(), "isWifiEnabled: " + isWifiEnabled);
```
Comment (sladomic) [inline app/src/main/java/org/literacyapp/appstore/service/ApplicationSynchronizationService.java]: Should this be here..

File: app/src/main/java/org/literacyapp/appstore/task/DownloadApplicationsAsyncTask.java [hunk 3]
```diff
@@ -52,11 +52,15 @@ public DownloadApplicationsAsyncTask(Context context) {
     protected Void doInBackground(Object... objects) {
         logger.info("doInBackground");
 
+        boolean isWifiEnabled = ConnectivityHelper.isWifiEnabled(context);
+        logger.info("isWifiEnabled: " + isWifiEnabled);
```
Comment (sladomic) [inline app/src/main/java/org/literacyapp/appstore/task/DownloadApplicationsAsyncTask.java]: ..or here?

File: app/src/main/java/org/literacyapp/appstore/service/ApplicationSynchronizationService.java (hunk 2 above)
Comment (sladomic) [inline app/src/main/java/org/literacyapp/appstore/service/ApplicationSynchronizationService.java]: The same for isWifiConnected

Comment (jogrimst) [inline app/src/main/java/org/literacyapp/appstore/service/ApplicationSynchronizationService.java]: Cleaned up the code. Thanks!

**Pull request merged**